### PR TITLE
Autofocus as parameter

### LINKED ISSUE #23: AUTO_FOCUS_INTERVAL_MS
Hi @dlazaro66 

I find Zxing AUTO_FOCUS_INTERVAL_MS is 2000L (2s)

but this project AUTO_FOCUS_INTERVAL_MS setting 5000L (5s).

Can create a function to setting this value?

Thx

### PR DESCRIPTION
**Pull request by dlazaro66**
fixes #23

### CODE REVIEW

File: qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java [hunk 1]
```diff
@@ -97,6 +97,14 @@ public void stopCamera() {
     mCameraManager.stopPreview();
   }
 
+  public void setAutofocusInterval(long autofocusIntervalInMs) {
+    mCameraManager.setAutofocusInterval(autofocusIntervalInMs);
+  }
+
+  public void enableTorch(boolean isEnabled) {
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java]: ```setTorchEnabled(boolean enabled)````

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraConfigurationManager.java [hunk 2]
```diff
@@ -288,21 +287,21 @@ boolean getTorchState(Camera camera) {
     return false;
   }
 
-  void setTorch(Camera camera, boolean newSetting) {
+  void enableTorch(Camera camera, boolean isEnabled) {
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraConfigurationManager.java]: ```setTorchEnabled(boolean enabled)```

Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraConfigurationManager.java]: Why package-private?

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraConfigurationManager.java [hunk 3]
```diff
@@ -288,21 +287,21 @@ boolean getTorchState(Camera camera) {
...
+  private void enableTorch(Camera.Parameters parameters, boolean isEnabled, boolean safeMode) {
+    enableTorch(parameters, isEnabled);
 
     if (!safeMode) {
-      setBestExposure(parameters, newSetting);
+      setBestExposure(parameters, isEnabled);
     }
   }
 
-  public static void setTorch(Camera.Parameters parameters, boolean on) {
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraConfigurationManager.java]: setTorchEnabled(boolean enabled)

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java [hunk 4]
```diff
@@ -72,6 +72,12 @@
     autoFocusAgainLater();
   }
 
+  public void setAutofocusInterval(long autofocusIntervalMs) {
+    if (autofocusIntervalMs > 0) {
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java]:
```java
if (autofocusIntervalMs <= 0) {
   throw new IllegalArgumentException("AutoFocusInterval must be greater than 0.");
}
```

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java [hunk 5]
```diff
@@ -44,8 +43,9 @@
 
   private static final String TAG = AutoFocusManager.class.getSimpleName();
 
-  private static final long AUTO_FOCUS_INTERVAL_MS = 5000L;
+  protected static final long DEFAULT_AUTO_FOCUS_INTERVAL_MS = 5000L;
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java]: private

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java [hunk 6]
```diff
@@ -137,20 +142,20 @@ public synchronized void setManualCameraId(int cameraId) {
...
         boolean wasAutoFocusManager = autoFocusManager != null;
         if (wasAutoFocusManager) {
           autoFocusManager.stop();
           autoFocusManager = null;
         }
-        configManager.setTorch(theCamera.getCamera(), newSetting);
+        configManager.setTorchEnabled(theCamera.getCamera(), enabled);
         if (wasAutoFocusManager) {
-          autoFocusManager = new AutoFocusManager(context, theCamera.getCamera());
+          autoFocusManager = new AutoFocusManager(theCamera.getCamera());
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]: why setting the torch on/off requires a new AutoFocusManager?

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java [hunk 7]
```diff
@@ -183,7 +188,8 @@ public synchronized void startPreview() {
     if (theCamera != null && !previewing) {
       theCamera.getCamera().startPreview();
       previewing = true;
-      autoFocusManager = new AutoFocusManager(context, theCamera.getCamera());
+      autoFocusManager = new AutoFocusManager(theCamera.getCamera());
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]: why start previewing requires a new AutoFocusManager?

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java (hunk 5 above)
Comment (dlazaro66) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java]: I need this protected, to access this default value from other classes in the same package. 
For example: https://github.com/dlazaro66/QRCodeReaderView/pull/64/files#diff-63e2aacd2af702b6dfe82ecf928463efR52

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java (hunk 7 above)
Comment (dlazaro66) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]: This is how original Zxing project is doing it. It seems like you need to create a new AutoFocusManager everytime you change the configuration.

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java (hunk 6 above)
Comment (dlazaro66) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]: This is how original Zxing project is doing it. It seems like you need to create a new AutoFocusManager every time you change the configuration.

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java [hunk 8]
```diff
@@ -71,6 +72,10 @@ public void setDisplayOrientation(int degrees) {
     }
   }
 
+  public void setAutofocusInterval(long autofocusIntervalInMs) {
+    this.autofocusIntervalInMs = autofocusIntervalInMs;
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]:
```java
if (autoFocusManager != null) {
   autoFocusManager.setAutofocusInterval(autofocusIntervalInMs);
}
```

File: qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java [hunk 9]
```diff
@@ -97,6 +97,14 @@ public void stopCamera() {
     mCameraManager.stopPreview();
   }
 
+  public void setAutofocusInterval(long autofocusIntervalInMs) {
+    mCameraManager.setAutofocusInterval(autofocusIntervalInMs);
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java]: Could mCameraManager ever be null at this point?

File: qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java [hunk 10]
```diff
@@ -97,6 +97,14 @@ public void stopCamera() {
...
+  }
+
+  public void setTorchEnabled(boolean enabled) {
+    mCameraManager.setTorchEnabled(enabled);
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/dlazaro66/qrcodereaderview/QRCodeReaderView.java]: Could mCameraManager ever be null at this point?

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java (hunk 5 above)
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java]: 👍

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java [hunk 11]
```diff
@@ -72,6 +72,13 @@
     autoFocusAgainLater();
   }
 
+  public void setAutofocusInterval(long autofocusIntervalMs) {
+    if (autofocusIntervalMs <= 0) {
+      throw new IllegalArgumentException("AutoFocusInterval must be greater than 0.");
+    }
+      this.autofocusIntervalMs = autofocusIntervalMs;
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/AutoFocusManager.java]: Fix alignment

File: qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java [hunk 12]
```diff
@@ -71,6 +72,12 @@ public void setDisplayOrientation(int degrees) {
     }
   }
 
+  public void setAutofocusInterval(long autofocusIntervalInMs) {
+    if (autoFocusManager != null) {
```
Comment (saguinav) [inline qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/CameraManager.java]: You forgot the:

```java
this. autofocusIntervalInMs = autofocusIntervalInMs;
```

**Pull request merged**